src/main/java/org/broadinstitute/hellbender/engine/HaploidWalker.java

```diff
@@ -70,6 +70,17 @@
      */
     protected abstract boolean allowsCheckOnly();
 
+    // the variant converter for get the haplotypes
+    private VariantHaplotypeConverter converter;
+
+    /** If override, this method should be called anyway for initialize the coverter. */
```

Comment:
Better handling of this. Probably the `haplotypeModelArgumentCollection` should have a `validateArgs` method for checking.